test/sys/test_signalfd.rs

```diff
@@ -0,0 +1,52 @@
+#[cfg(target_os = "linux")]
+#[test]
+// FIXME: Fails with "fatal runtime error: failed to initiate panic, error 5" on i686-unknown-linux-musl
+#[cfg_attr(all(target_env = "musl", target_arch = "x86"), ignore)]
+fn test_signalfd() {
+    use nix::sys::signalfd::{SFD_NONBLOCK, SignalFd};
+    use nix::sys::signal;
+    use nix::unistd;
+    use nix::unistd::*;
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+
+    // Set up a signalfd to listen for SIGUSR1 on this thread
+    let mut mask = SigSet::empty();
+    mask.add(SIGUSR1);
+    mask.thread_block().unwrap();
+
+    // Start a child process
+    match unistd::fork() {
```

Comment:
I don't think you should fork here.  For one thing, cargo's test harness has deadlocks when a multithreaded test program forks.  That's why I had to add std::process:exit(0) all over the place.  For another thing, it's not necessary.  You can just create the eventfd and send the signal all from the same thread, just like the old test did.